### PR TITLE
feat: add a checkbox to select all items in session list

### PR DESCRIPTION
**Pull request by ironAiken2**
### TL;DR

Add functionality to select multiple sessions using a header checkbox in the BackendAI session list.

### What changed?

- Introduced a new property `_boundHeaderCheckboxRenderer` to bind the header checkbox renderer function.
- Added a new property `sessionAllChecked` to track the state of the header checkbox.
- Implemented `_toggleSelectAll` method to handle the select/deselect all functionality.
- Added `headerCheckboxRenderer` method to render the header checkbox.

### How to test?

1. Navigate to the session list in the BackendAI application.
2. Verify the presence of a header checkbox in the session list table.
3. Click the header checkbox to select all sessions; click again to deselect.
4. Ensure the multi-action buttons are appropriately displayed based on selection.

### Why make this change?

This change enhances the session management capabilities by allowing users to quickly select and manage multiple sessions at once, improving usability and efficiency.

---

**Checklist:** (if applicable)

- [ ] Mention to the original issue
- [ ] Documentation
- [ ] Minium required manager version
- [ ] Specific setting for review (eg., KB link, endpoint or how to setup)
- [ ] Minimum requirements to check during review
- [ ] Test case(s) to demonstrate the difference of before/after

### CODE REVIEW

**Comment by ironAiken2** [comment]
* **#2519** <a href="https://app.graphite.dev/github/pr/lablup/backend.ai-webui/2519?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @ironAiken2 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: src/components/backend-ai-session-list.ts [hunk 1]
```diff
@@ -3955,6 +3958,39 @@ ${rowData.item[this.sessionNameField]}</pre
...
+      if (this._selected_items.length > 0) {
+        this.multipleActionButtons.style.display = 'block';
+      } else {
+        this.multipleActionButtons.style.display = 'none';
+      }
+    });
+    [...elm].forEach((checkbox) => {
+      checkbox.setAttribute('checked', 'true');
+    });
+    console.log(this._selected_items);
```
Comment (agatha197) [inline src/components/backend-ai-session-list.ts]:
```suggestion
```

**Review by agatha197 — CHANGES_REQUESTED** [review]
Is it possible to make the header checkbox checked when all items are selected, and unchecked when all items are unchecked? Or you can use just vaadin selection column. ref: https://vaadin.com/docs/latest/components/grid#selection

**Comment by ironAiken2** [comment]
It was already built into vaadin library... why didn't I find it? I'll look into it and incorporate it. Thank you for your review @agatha197

**Review by yomybaby — CHANGES_REQUESTED** [review]
I found a bug. After a few seconds of selecting a session, the checkbox automatically changes to unchecked. Please check it.

File: src/components/backend-ai-session-list.ts [hunk 2]
```diff
@@ -4318,15 +4284,12 @@ ${rowData.item[this.sessionNameField]}</pre
         <vaadin-grid id="list-grid" theme="row-stripes column-borders compact dark" aria-label="Session list"
           .items="${this.compute_sessions}" height-by-rows>
           ${
-            this._isRunning
+            this._isRunning || this._APIMajorVersion > 4
```
Comment (agatha197) [inline src/components/backend-ai-session-list.ts]:
```suggestion
            this._isRunning
```
This code adds a selection column to the terminated session tabs. Let's remove it.

**Review by agatha197 — CHANGES_REQUESTED** [review]
LGTM But please reflect one feedback :)

**Comment by yomybaby** [comment]
LGTM!

**Review by yomybaby — APPROVED** [review]

**Review by agatha197 — APPROVED** [review]
LGTM

**Pull request merged**